pkg/ext/stores/tokens/tokens.go

```diff
@@ -1301,10 +1309,13 @@ func tokenFromSecret(secret *corev1.Secret) (*ext.Token, error) {
 	if err != nil {
 		return token, err
 	}
-	// inject default on retrieval
-	if ttl == 0 {
-		ttl = ThirtyDays
+
+	// clamp and inject default on retrieval
```

Comment:
I don't think we can "inject" the default on retrieval. The object should be immutable on read. The setting can change at any point in time resulting in a different outcome.
The default makes sense only on write - when creating and maybe when updating. Which means that the ternary (0<, 0, >0) representation of ttl in the backing secret doesn't make a lot of sense.

Reply:
The stuff on retrieval has the thought of `protect against shenigans due to somebody managing to edit the backing store directly` behind it, if I remember my own thoughts correctly. 

If we drop that idea then yes, it is much better, and likely easier, to perform default resolution and clamping only on create and update, and simply take the backing store as truth to not change on list, get, watch.

Reply:
done, a27d428eb5c09934b028d91d51882d612ad53f47
also, not sure what you looked at to still have `ThirtyDays` in it. That should have been gone in the PR.

Reply:
Yes. I am not sure why I was thinking that way.